### DISCUSSION TITLE
Possible to open with code instead of in browser tab?

### DISCUSSION THREAD

**Discussion by IngCr3at1on** [coder/code-server#2114] [Q&A] [answered]
I was previously an sshcode user and while I am in fact a go developer and felt like the app itself was not super complicated I can understand the reasons for deprecating it. I've been using code-server by itself for awhile now and it works fine but what I haven't figured out is how I can open a connection inside code itself instead of in a browser window. sshcode did this automatically so I believe this is still possible I'm just not sure how.

Is this in fact possible and can someone tell me how to do it if so (might be worth throwing in the FAQ if it is indeed possible)?

**Comment by code-asher** [comment, marked as answer]
I'm not 100% sure I understand the use case (I think this is related to the port forwarding part of sshcode?) but we haven't moved all the sshcode functionality into code-server yet so I think it'll be possible once that's complete: https://github.com/cdr/code-server/pull/1996

**Reply by IngCr3at1on** [reply]
I can't imagine what this would have to do with the port forwarding itself... I just want to open the connection in the vscode editor installed on my client instead of in a browser tab; I highly doubt this was related to the port-forwarding behavior but that was kind of the point of asking here...

I guess the better question is; how exactly did/does sshcode open a connection to a server inside of vscode instead of in a browser tab (IIRC this would have to be a feature of electron but that's pretty out of my wheelhouse)?

**Reply by code-asher** [reply]
What kind of connection are you trying to open in VS Code? Is this something like accessing a SQL server using the SQL extension from inside VS Code or something like that?

**Reply by IngCr3at1on** [reply]
@code-asher I really don't understand the confusion here...

As the sshcode documentation mentions running it will start a code-server on a remote host and open an editor connected to that host. If vscode is actually installed on that client machine it will open it within vscode instead of opening it in a browser tab...

I simply want to know how I can mimic that last thing without using sshcode... I already have a working code-server running on my remote host and I have vscode running on my client system but what I don't have is a way to open vscode on the client system where it's talking to code-server; instead I have to connect to the code-server using a browser tab...

I'm sorry if I appear frustrated in my response but I literally don't understand what your responses have to do with my question. In the simplest wording possible: I want to open a connection to code-server where it opens inside of vscode on my client system _instead_ of using my browse.

**Reply by code-asher** [reply]
Ahhhhhh I think I see the confusion. sshcode doesn't/can't open anything in your locally installed VS Code; everything is in a browser which is where I got lost.

It forwards a port to your remote server through SSH then it loads that port through localhost using a browser. If you have Chrome installed it opens it in app mode which removes all of the surrounding UI. That makes it look like a local app but it doesn't have anything to do with VS Code installed locally.

Our installation script will eventually do this for you I think but since you have it all up and running and accessible at a URL you can do this:

```
chrome --app=<code server url>
```

**Reply by IngCr3at1on** [reply]
That's what I was looking for; thanks!
